### PR TITLE
Add modal tests

### PR DESCRIPTION
**Pull request by kielbasa-elp**

### CODE REVIEW

File: packages/design-system/src/components/Modal/__tests__/Modal.screen-object.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    return this;
+  }
+
+  openModal() {
+    const button = screen.getByRole("button", { name: /open modal/i });
+
+    act(() => {
+      button.click();
+    });
+
```
Comment (kielbasa-elp) [inline packages/design-system/src/components/Modal/__tests__/Modal.screen-object.tsx:21]: This click manipulates state so it have to be wrapped by act

File: packages/design-system/src/components/Modal/__tests__/Modal.screen-object.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  get modal() {
+    return screen.queryByRole("dialog");
+  }
+
+  closeModal() {
+    const overlay = document.querySelector(".ReactModal__Overlay");
+    if (!overlay) throw new Error("Modal not opened");
+
+    act(() => {
```
Comment (zuzannanap) [inline packages/design-system/src/components/Modal/__tests__/Modal.screen-object.tsx:33]: https://plainenglish.io/blog/you-probably-dont-need-act-in-your-react-tests-2a0bcd2ad65c

Comment (zuzannanap) [reply packages/design-system/src/components/Modal/__tests__/Modal.screen-object.tsx:33]: we can use userEvent.click(element) without wrapping in act

File: packages/design-system/src/components/Modal/__tests__/Modal.screen-object.tsx (hunk 1 above)
Comment (zuzannanap) [reply packages/design-system/src/components/Modal/__tests__/Modal.screen-object.tsx:21]: not when you use userEvent

File: packages/design-system/src/components/Modal/__tests__/Modal.screen-object.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      (overlay as HTMLElement).click();
+    });
+
+    return this;
+  }
+}
+
+const TestModal = ({ ...props }: Partial<ModalProps>) => {
```
Comment (zuzannanap) [inline packages/design-system/src/components/Modal/__tests__/Modal.screen-object.tsx:41]: TestModalComponents - to be consistent

File: packages/design-system/src/components/Modal/__tests__/Modal.screen-object.tsx (hunk 2 above)
Comment (zuzannanap) [reply packages/design-system/src/components/Modal/__tests__/Modal.screen-object.tsx:33]: i think it's more readable

**Review by zuzannanap — APPROVED** [review]

File: packages/design-system/src/components/Modal/__tests__/Modal.screen-object.tsx (hunk 3 above)
Comment (zuzannanap) [inline packages/design-system/src/components/Modal/__tests__/Modal.screen-object.tsx:41]: TestModalComponent to be consistent

**Review by zuzannanap — CHANGES_REQUESTED** [review]

**Pull request merged**